### PR TITLE
updated Bio.Restriction for python3

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request updates Bio.Restriction for python3. Travis, don't disappoint me.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
GitHub won't show me as there were so many changes, but you've updated ``Bio/Restriction/Restriction_Dictionary.py`` which is an automatically generated file.

Did you mean to commit ``Scripts/Restriction/ranacompiler.py`` and/or ``Scripts/Restriction/rebase_update.py`` as well?

**Comment by mdehoon** [comment]
@peterjc  I was not aware of those scripts.

**Comment by peterjc** [comment]
So you've fixed the autogenerated code to be black compliant (and got rid of the old Jython hack with the ``_temp`` functions - see also #2551 for some similar work)? Nice.

I think I (or Markus) need to look at the diff at the command line to review this (since GitHub won't show such a large diff in the browser).

**Comment by peterjc** [comment]
I propose to separate the REBASE update from the style work, to make the changes easier to review.

To that end, I can make a new pull request starting from the master, just updating REBASE.

I can then rebase this pull request on top of it - can get just the style changes.

File: Scripts/Restriction/ranacompiler.py [hunk 1]
```diff
@@ -367,9 +366,11 @@ def __init__(cls):
...
 # as part of this package.
 #
-# This file is automatically generated - do not edit it by hand! Instead,
-# use the tool Scripts/Restriction/ranacompiler.py which in turn uses
-# Scripts/Restriction/rebase_update.py and Bio/Restriction/RanaConfig.py
+# THIS FILE IS AUTOMATICALLY GENERATED - DO NOT EDIT IT BY HAND!
+#
+# Instead, use the tool Scripts/Restriction/ranacompiler.py which in turn
+# uses Scripts/Restriction/rebase_update.py and Bio/Restriction/RanaConfig.py
+
```
Comment (MarkusPiotrowski) [inline Scripts/Restriction/ranacompiler.py]: This line should be shortened to:
```python
# uses Scripts/Restriction/rebase_update.py
```

`RanaConfig.py` was deleted two or three releases ago.

Comment (peterjc) [reply Scripts/Restriction/ranacompiler.py]: Good catch - I'll fix that on #2592, and then update this pull request accordingly.

**Comment by peterjc** [comment]
Thanks Markus - I need to deal with the conflict so will rebase this after #2592 to exclude those files.

**Comment by peterjc** [comment]
Rebased and autogenerated files changes removed (done in #2592). OK to merge @mdehoon?

**Pull request merged**